packages/generators/init-generator.ts

```diff
@@ -40,16 +43,63 @@ export default class InitGenerator extends Generator {
 		super(args, opts);
 		this.isProd = false;
 		this.usingDefaults = false,
+
 		this.dependencies = [
 			"webpack",
 			"webpack-cli",
-			"terser-webpack-plugin",
+			"webpack-dev-server",
 			"babel-plugin-syntax-dynamic-import",
 		];
+		if (this.isProd) {
+			this.dependencies.push("terser-webpack-plugin");
+		}
+
 		this.configuration = {
 			config: {
 				topScope: [],
-				webpackOptions: {},
+				// tslint:disable: object-literal-sort-keys
+				webpackOptions: {
+					mode: this.isProd ? "'production'" : "'development'",
+					entry: undefined,
+					output: {},
+					module: {
+						rules: [],
+					},
+					plugins: this.isProd ? [] : getDefaultPlugins(),
+				},
+				// tslint:enable: object-literal-sort-keys
+			},
+		};
+
+		// add splitChunks options for transparency
+		// defaults coming from: https://webpack.js.org/plugins/split-chunks-plugin/#optimization-splitchunks
+		this.configuration.config.topScope.push(
+			"const path = require('path')",
+			"\n",
+			tooltip.splitChunks(),
+		);
+
+		if (this.isProd) {
+			this.configuration.config.topScope.push(
+				tooltip.terser(),
+				"const TerserPlugin = require('terser-webpack-plugin');",
+				"\n",
+			);
+		}
+
+		this.configuration.config.webpackOptions.optimization = {
```

Comment:
I'm in favor for using `splitChunks: all`. Works good as a initial default, at least in dev mode

Reply:
Only in dev mode is fine as long as we use the in memory output (that's what dev server does). But not in production, we can't assume people have the correct version of `HtmlWebpackPlugin` with the correct configuration

Reply:
Thanks! So, I'm going to keep it in production and using "all" in dev mode, let me know if it should be done differently